lib/route.js

```diff
@@ -116,7 +116,9 @@ function buildRouting (options) {
   }
 
   // Route management
-  function route (opts) {
+  function route (options) {
+    const opts = { ...options }
```

Comment:
This is a shallow clone. Shouldn't `rfdc` be used here?

Reply:
I thought that a shallow one should be enugh, buy maybe we need a deep clone because of this https://github.com/fastify/fastify/blob/7369e7e3f0e881afecb5cf8854afb03edc708b83/lib/route.js#L185

Reply:
In fact, isn't those line the issue: https://github.com/fastify/fastify/blob/master/lib/route.js#L176-L179 ?
We are mutating the option object.

Reply:
Right. The issue is that properties of the passed in options object get mutated, thereby mutating the original passed by reference object. To solve the issue, each route should have its own copy of the passed in object.

Reply:
Yes, but cloning the initial opts, we are fixing this issue. It is ok now to assign to opts since it is locally scoped to that closure. I am still not sure that a deep clone is needed, though.

Reply:
I don't know enough. If you say this is the solution, then it is the solution.

👍

Reply:
Based on those lines, we are mutating/assigning only first level props. I think for this specific mutation it is fine to have a shallow copy using the spread operator. But of course, I could be wrong :)

Reply:
In this case shallow copy is OK because regarding the functions and object you're already re-using the route object, so you assume that those are references and not copy. It's ok to me.

Reply:
@fox1t Can you please add a comment with the explanation you did in: https://github.com/fastify/fastify/pull/2569#discussion_r492007037?

Reply:
Done